### PR TITLE
new packages: libbacktrace, libpfm, numap, numamma

### PR DESCRIPTION
**Pull request by trahay**
Hi,
Numamma (https://numamma.github.io/numamma/) is a memory profiler for parallel applications. It depends on several other projects (libbacktrace, libpfm, numap) that may be useful for other HPC projects, so I created 4 spack packages.

François

### CODE REVIEW

File: var/spack/repos/builtin/packages/libpfm/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "https://github.com/wcohen/libpfm4"
+    git      = "https://github.com/wcohen/libpfm4.git"
+    maintainers = ['trahay']
+
+    version('master',  branch='master')
+    version('2020-03-09', commit='210b2ef95f33eccb671f2a88a979de5364c94465')
+
+    def install(self, spec, prefix):
+        env['PREFIX'] = prefix
+        make('install')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpfm/package.py:21]: I would replace this whole thing with:
```python
def setup_build_environment(self, env):
    env.set('PREFIX', self.prefix)
```

**Comment by mwkrentel** [comment]
@trahay There already is a libpfm4 package from sourceforge.
Does that not work for you?
Is this a custom fork or something?

@adamjstewart  Btw, what is the policy on adding a custom fork to spack?
What should you name the package, maybe `<orig-package>-<my-label>` ?

I don't need to do this today, but there have been times where I've
thought about adding a separate fork.  Something with too many patches
or modified API that wouldn't fit in the original package.

**Comment by adamjstewart** [comment]
@mwkrentel it depends on whether or not the fork has taken on a life of its own. If it's just a simple fork with a few hacks to add support for something, it can be added to the same package as an additional version. If it's become a completely separate software product with a larger community (like OpenFOAM) then we make a new package.

In the past, the policy was that if `spack create` automatically detected the package name, and it was a different package name, it would get a different package. That's why Python 2 and 3 have a single package but Anaconda/Miniconda 2 and 3 have different packages. If the authors of the package think their software is different enough to warrant a different name, we don't argue.

We've also compared Spack to other package managers when making such decisions in the past.

**Comment by trahay** [comment]
> @trahay There already is a libpfm4 package from sourceforge.
> Does that not work for you?

Oups, I didn't see this package ! Here's a new commit that removes my libpfm package.

> Is this a custom fork or something?

I'm not sure about the status of the github repo I used. It seems that the official repo is on sourceforge.net, but the github repo contains lots of recent commits from Stephane Eranian (who is the main contributor recently on sourceforge.net).
Anyway, let's use the official one.

**Pull request merged**